### ISSUE TITLE
Bright Grey Background on High Sierra with Multiple Displays

### ISSUE THREAD

**Issue by timkite** [volendavidov/NagBar#4]
The 1.3.2 version does fix the permanent bright grey background on the NagBar menu item when a High Sierra Mac has a single display. It does not, however, if the Mac has more than one display. Upon launch the app will display normally, but as soon as a secondary display is made active (by opening an app there or just clicking on its desktop), the NagBar menu item will display a bright grey background on the active display. It will display properly on the inactive display, and this behavior will follow the active display. If you'd like more screenshots or logs, let me know.

I reported this via email before remembering you preferred GitHub reports, so I apologize for double reporting.
[Screenshots - High Sierra Dual Display.zip](https://github.com/volendavidov/NagBar/files/1349170/Screenshots.-.High.Sierra.Dual.Display.zip)

**Comment by timkite** [comment]
I double-checked, and the problem is there with dark or light mode enabled, and I made doubly sure I'm running 1.3.2. Configuration is a 5K iMac as primary display and a 23" Cinema Display (yeah, vintage) as secondary display to the left, Mac OS 10.13 (17A365). Once I get a chance after work I'll try with my laptop (13" Retina MacBook Pro, 2015 vintage) using the Cinema Display as a secondary and see if the same is true there as well.

**Comment by yveslaroche** [comment]
I have the same issue and I'm also using two external monitors.

**Comment by volendavidov** [comment]
Can you test if this fixes it? If not, there is another workaround that I can use, though a bit uglier.
[NagBar.zip](https://github.com/volendavidov/NagBar/files/1360673/NagBar.zip)

**Comment by timkite** [comment]
That one has an interesting new behavior. Initially, there's no menu bar entry for NagBar on the main display, but it appears normal on the secondary monitor. Then, when switching to secondary, the "bright grey" background behavior resumes on the active screen as before. I'm planning on trying on the update to High Sierra that was just released.

I also tried on my laptop earlier today with a projector and the behavior there is interesting as well. Even though the projector is mirrored, it acts like a dual display in that it shows the grey background on both the laptop and the projector.

**Comment by timkite** [comment]
Alright, just tried the latest High Sierra update (10.13 17A405) and there's no change. 1.3.2 starts up normally, then goes bright grey on the active screen once the active screen is switched once. Your new patch still starts off not displaying on the primary screen and displaying normally on the secondary screen, then shows a grey background on the active screen once the active screen is switched.

Aren't new OS releases fun? I wonder if iStat Menus is running into this same issue.

**Comment by volendavidov** [comment]
Thanks for the testing. I have never had an OS release that is as "fun" as this one :). I don't know about iStat Menus, but I noticed a similar problem with 1Password, when you click on it.

Anyway, I have prepared a new version where I think it should be OK. It won't be OK when "shake" animation is selected though (basically the status bar animations are what causes the problem) so I will either remove it as an option, leave it with a notice or perhaps find a way to fix it as well.
[NagBar.zip](https://github.com/volendavidov/NagBar/files/1366980/NagBar.zip)
